In [1]:
import pandas as pd

In [2]:
data = pd.read_csv("School_Neighborhood_Poverty_Estimates_-_Current.csv")
data =data.head(1000)
data['LAT'] = data['LAT'].astype(int)
X = data[['IPR_EST','IPR_SE']]
y = data['LAT']

In [3]:
import matplotlib.pyplot as plt
from sklearn import svm

C=1.0
# svc with linear function
svc =svm.SVC(kernel='linear', C=C).fit(X,y)
# Linearsvc
lin_svc =svm.LinearSVC(C=C).fit(X,y)
# svc with RBF kernel
rbf_svc =svm.SVC(kernel='rbf', gamma=0.7, C=C).fit(X,y)
# svc with polynomial kernel for degree>1
poly_svc=svm.SVC(kernel='poly',degree=3, C=C).fit(X,y)

C:\Users\Aravind Kaspe\Anaconda 3\Lib\site-packages\sklearn\svm\_classes.py:32: FutureWarning: The default value of `dual` will change from `True` to `'auto'` in 1.5. Set the value of `dual` explicitly to suppress the warning.
  warnings.warn(
C:\Users\Aravind Kaspe\Anaconda 3\Lib\site-packages\sklearn\svm\_base.py:1242: ConvergenceWarning: Liblinear failed to converge, increase the number of iterations.
  warnings.warn(


In [4]:
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.svm import SVC, LinearSVC
from sklearn.metrics import accuracy_score
import pandas as pd
import numpy as np

# Convert data to DataFrame
df = data

# Selecting features and target variable
X = data[['IPR_EST','IPR_SE']]
y = data['LAT']

# Splitting the dataset into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.4, random_state=42)

# Feature scaling
scaler = StandardScaler()
X_train_scaled = scaler.fit_transform(X_train)
X_test_scaled = scaler.transform(X_test)

# Training and evaluating models
models = {
    'Linear' : SVC(kernel='linear'),
    'LinearSVC': LinearSVC(max_iter=10000),
    'SVC (RBF Kernel)': SVC(kernel='rbf'),
    'SVC (Polynomial Kernel)': SVC(kernel='poly')
}

# Dictionary to hold accuracy scores
accuracy_scores = {}

for name, model in models.items():
    model.fit(X_train_scaled, y_train)
    y_pred = model.predict(X_test_scaled)
    accuracy_scores[name] = accuracy_score(y_test, y_pred)

accuracy_scores

C:\Users\Aravind Kaspe\Anaconda 3\Lib\site-packages\sklearn\svm\_classes.py:32: FutureWarning: The default value of `dual` will change from `True` to `'auto'` in 1.5. Set the value of `dual` explicitly to suppress the warning.
  warnings.warn(


{'Linear': 0.3625,
 'LinearSVC': 0.3475,
 'SVC (RBF Kernel)': 0.37,
 'SVC (Polynomial Kernel)': 0.3375}

Interpretation: 

The above model prediction shows the accuracy scores for different models trained to predict the latitude ('LAT') of a location based on 'IPR_EST' and 'IPR_SE'. The models evaluated include:

Linear SVC with a linear kernel: Achieved an accuracy score of 0.3625.

LinearSVC: Achieved an accuracy score of 0.3475.

SVC with RBF (Radial Basis Function) kernel: Achieved the highest accuracy score of 0.37.

SVC with Polynomial kernel: Had the lowest accuracy score of 0.3375.

These accuracy scores reflect the proportion of correctly predicted latitudes out of all predictions made by each model on the test set. The scores are relatively close to each other, suggesting that there isn't a significant difference in the performance of these models for this particular task, under the configurations and dataset used.